### PR TITLE
Listing reorg Chapter 1 to 7

### PR DESCRIPTION
**Pull request by TAdev0**
+ bump cairo version
+ reorg listings and file names
+ update the  ''✅ Compiles'' in the module chapter

This change is [Reviewable](https://reviewable.io/reviews/cairo-book/cairo-book/685)

### CODE REVIEW

**Comment by TAdev0** [comment]
@enitrat  ready for review

this concerns chapter 1 to 7.

Will do others in another PR.

**Comment by TAdev0** [comment]
Also updated `find_value_iterative` function in Enum chapter to use while let now it is available

**Comment by enitrat** [comment]
> 
Also updated find_value_iterative function in Enum chapter to use while let now it is available

**Comment by TAdev0** [comment]
No

**Review by enitrat — APPROVED** [review]
I don't think this should have been done for every single one, it doesn't really matter if there's an off-by-one count in the name of the listing file being referred in the book. The only important thing was that they lived in the correct chapters

File: src/ch02-03-functions.md [hunk 1]
```diff
@@ -121,24 +121,20 @@ of functions.
...
-assigning a value to it with the `let` keyword is a statement. In Listing {{#ref fn-main}},
+assigning a value to it with the `let` keyword is a statement. In the following `main` function,
 `let y = 6;` is a statement.
 
 ```rust
-{{#include ../listings/ch02-common-programming-concepts/listing_01_statement/src/lib.cairo}}
+{{#include ../listings/ch02-common-programming-concepts/no_listing_19_statement/src/lib.cairo}}
 ```
 
-{{#label fn-main}}
```
Comment (enitrat) [inline src/ch02-03-functions.md:131]: restore this listing

**Review by enitrat — CHANGES_REQUESTED** [review]

File: src/ch02-03-functions.md (hunk 1 above)
Comment (TAdev0) [reply src/ch02-03-functions.md:131]: done

**Pull request merged**